pydm/widgets/archiver_time_plot.py

```diff
@@ -816,6 +831,7 @@ def createCurveItem(self, *args, **kwargs) -> ArchivePlotCurveItem:
         """Create and return a curve item to be plotted"""
         curve_item = ArchivePlotCurveItem(*args, **kwargs)
         curve_item.archive_data_received_signal.connect(self.archive_data_received)
+        curve_item.prompt_archive_request.connect(self.requestDataFromArchiver)
```

Comment:
The `prompt_archive_request` signal has already been emitted on line 113 above, but will get ignored if the pv address is passed as part of initialization of the curve item since this signal/slot connection here isn't made until after it was already emitted as part of the init. Will want to move this so the initial request isn't lost in some cases.

Reply:
That is a good point. I think this was a non-issue for the way I use ArchiveCurveItems in the Archive Viewer, since I create an object without an address and set the address after the fact. I agree with you this should probably be changed.

What do you think about passing these slots into the initializer and making the connections in `ArchiveCurveItem.__init__`? Are there other ways to establish these connections before initialization?

Reply:
Yeah I think that sounds fine to me. The other way is maybe I should have had more of that request data logic originally in the `ArchivePlotCurveItem` itself rather than the plot object but that's probably beyond the scope of this PR and can be looked at later if needed.

Reply:
Wait, instead of passing the slots into the initializer to establish a connection there, what if I just called `self.requestDataFromArchiver` after this line? 🤦 
That's essentially what I'm trying to do for anyway, and making the connection in the initializer seems to be causing problems. Trying it out now.

Reply:
Added `self.requestDataFromArchiver()` to `PyDMArchiverTimePlot.addYChannel()`